webpack/assets/javascripts/pages/statistics_page.js

```diff
@@ -0,0 +1,42 @@
+import donut from '../charts/c3DonutChartService';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import StatisticsChartsList from "../react_app/components/charts/StatisticsChartsList";
+
+export function generateChart(details) {
+    let dataEventHandlers = {
+        onclick: getClickHandler(details.url)
+    };
+
+    let config = donut.getDonutConfig(details.data, '#' + details.id + 'Chart', dataEventHandlers);
+
+    donut.generate(config);
+
+    function getClickHandler(url) {
+        return function (d, i) {
+            var val = '"' + encodeURIComponent(d.name) + '"';
+            if (url.indexOf("~VAL1~") != -1) {
+                url = url.replace("~VAL1~", val);
```

Comment:
please note that val in this case is not the current string to replace. 
for example, in my chart the value was "2 cores",  yet the value to use should have been only "2" (it simply generate the wrong search query)